contracts/auctions/FungibleAuction.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.3;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+// TODO: Use interface. The current version of @openZeppelin/contracts lib has none that fits.
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+// TODO: Switch to proper interface?
+interface IAuctionHouse {
+    function closeFungible(
+        address buyTokenAddress,
+        address sellTokenAddress,
+        uint256 tokensSold,
+        uint256 minPrice
+    ) external;
+}
+
+/// @title Auction for fungible tokens
+contract FungibleAuction {
+    using SafeMath for uint256;
+
```

Comment:
Why not using SafeERC20 for IERC20; and switching to safe transfers?

Reply:
Laziness. I agree that adding the `SafeERC20` lib for this is better. Also `IERC20` interface instead of the `ERC20` contract is better. Will change.

There used to be a check using `decimals` which isn't on the `IERC20` interface, hence why I changed it at some point. Or lack of coffee, who knows?